src/appleseed/foundation/platform/path.cpp

```diff
@@ -133,6 +135,49 @@ const char* get_executable_directory()
     return path;
 }
 
+const char* get_home_directory()
+{
+    static char path[FOUNDATION_MAX_PATH_LENGTH + 1];
+    static bool path_initialized = false;
+
+    if (!path_initialized)
+    {
+#if defined _WIN32
+
+        return 0;
+
++#elif defined __APPLE__
+
+        return 0;
+
+#elif defined __linux__
+
+        const char *home_dir = getenv("HOME");
+
+        if (home_dir == 0)
+            home_dir = getpwuid(getuid())->pw_dir;
+
+        assert(home_dir);
+        strcpy(path, home_dir);
```

Comment:
@est77 All the functions exist on FreeBSD, but of course that doesn't mean this is the correct way to retrieve the home directory. If you're not sure, let the FreeBSD case unimplemented and maybe @danfe can extend your implementation to FreeBSD if it turns out to be the correct way to do it?